views/index.js

```diff
@@ -0,0 +1,32 @@
+fetch("http://news-summary-api.herokuapp.com/guardian?apiRequestUrl=http://content.guardianapis.com/search?show-fields=thumbnail&show-elements=image")
+  .then(response => {
+    console.log("response:")
+    console.log(response)
+    return response.json()
+  })
+  .then(data => {
+    console.log("data:")
+    console.log(data)
+    let subHeading = document.getElementById("subHeading")
+    data.response.results.forEach((article, index) => {
+      subHeading.insertAdjacentHTML('afterend', `<br>`)
+      subHeading.insertAdjacentHTML('afterend', `<img src="${data.response.results[index].fields.thumbnail}"><br>`)
+      subHeading.insertAdjacentHTML('afterend', `<a href="#${index}">${data.response.results[index].webTitle}</a><br>`)
+    })
+    return data;
+  })
+  .then(data => {
+    window.addEventListener("hashchange", function() {
+      console.log("hashchange detected")
+      let index = location.hash.split('#')[1];
+      let headline = data.response.results[index].webTitle
```

Comment:
Just wonder if it's worth assigning data.response.results to a variable so you don't keep having to use it. Although, other than reading it, I'm not sure it would really make a difference to be honest 🤷‍♀️